docx/text/paragraph.py

```diff
@@ -62,6 +62,36 @@ def clear(self):
         self._p.clear_content()
         return self
 
+    @property
+    def footnotes(self):
+        """
+        Returns a list of |Footnote| instances that refers to the footnotes in this paragraph,
+        or |None| if none footnote is defined.
+        """
+        reference_ids = self.footnote_reference_ids
+        if reference_ids == None:
+            return None
+        footnotes = self._parent._parent.footnotes
+        footnote_list = []
+        for ref_id in reference_ids:
+            footnote_list.append(footnotes[ref_id])
+        return footnote_list
+
+    @property
+    def footnote_reference_ids(self) -> (list[int]|None):
+        """
+        Return all footnote reference ids (``<w:footnoteReference>``) form the paragraph,
+        or |None| if not present.
+        """
```

Comment:
Similar comment as for `Run`, should we expose this API on higher level? Maybe moving this to lower level API, and here we just expose API for managing footnotes and not their ids?